docs/source/basics/prompt_templates.rst

```diff
@@ -0,0 +1,249 @@
+        path="/tmp/Mistral-7B-v0.1/tokenizer.model"
+        prompt_template="torchtune.models.mistral.MistralChatTemplate"
+    )
+
+.. code-block:: yaml
+
+    # In config
+    tokenizer:
+      _component_: torchtune.models.mistral.mistral_tokenizer
+      path: /tmp/Mistral-7B-v0.1/tokenizer.model
+      prompt_template: torchtune.models.mistral.MistralChatTemplate
+
+
+Defining via dictionary
+^^^^^^^^^^^^^^^^^^^^^^^
+
+For example to achieve the following prompt template:
+
+.. code-block:: text
+
+    System: {content}\\n
+    User: {content}\\n
+    Assistant: {content}\\n
+    Tool: {content}\\n
+
+You need to pass in a tuple for each role, where ``PREPEND_TAG`` is the string
+added before the text content and ``APPEND_TAG`` is the string added after.
+
+.. code-block:: python
+
+    template = {role: (PREPEND_TAG, APPEND_TAG)}
+
+Thus, the template would be defined as follows:
+
+.. code-block:: python
+
+    template = {
+        "system": ("System: ", "\\n"),
+        "user": ("User: ", "\\n"),
+        "assistant": ("Assistant: ", "\\n"),
+        "ipython": ("Tool: ", "\\n"),
+    }
+
+Now we can pass it into the tokenizer as a dictionary:
+
+.. code-block:: python
+
+    # In code
+    from torchtune.models.mistral import mistral_tokenizer
+
+    template = {
+        "system": ("System: ", "\\n"),
+        "user": ("User: ", "\\n"),
+        "assistant": ("Assistant: ", "\\n"),
+        "ipython": ("Tool: ", "\\n"),
+    }
+    m_tokenizer = mistral_tokenizer(
+        path="/tmp/Mistral-7B-v0.1/tokenizer.model"
+        prompt_template=template,
+    )
+
+.. code-block:: yaml
+
+    # In config
+    tokenizer:
+      _component_: torchtune.models.mistral.mistral_tokenizer
+      path: /tmp/Mistral-7B-v0.1/tokenizer.model
+      prompt_template:
+        system:
+          - "System: "
+          - "\\n"
+        user:
+          - "User: "
+          - "\\n"
+        assistant:
+          - "Assistant: "
+          - "\\n"
+        ipython:
+          - "Tool: "
+          - "\\n"
+
+If you don't want to add a prepend/append tag to a role, you can just pass in an empty string "" where needed.
+
+Using the :class:`~torchtune.data.PromptTemplate` class
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+A template dictionary can also be passed into :class:`~torchtune.data.PromptTemplate` so you can use it as a standalone custom
+prompt template class.
+
+.. code-block:: python
+
+    from torchtune.data import PromptTemplate
+    from functools import partial
+
+    # As a builder function
+    def my_custom_template() -> PromptTemplate:
+        return PromptTemplate(
+            template={
+                "user": ("User: ", "\\n"),
+                "assistant": ("Assistant: ", "\\n"),
+            },
+        )
+
+    # As a class
+    MyCustomTemplate = partial(
+        PromptTemplate,
+        template={
+            "user": ("User: ", "\\n"),
+            "assistant": ("Assistant: ", "\\n"),
+        },
+    )
```

Comment:
Imo this is like 25% too slick for a docs page

Reply:
fine